go/module.hlb

```diff
@@ -103,90 +148,168 @@ string builderImage(variadic option::template args) {
 	{{- end -}}
 	END with option {
 		stringField "base" "docker.elastic.co/beats-dev/golang-crossbuild"
-		defaultArgs
-		args
+		defaultTmplOpts
+		tmplOpts
 	}
 }
 
-option::template defaultArgs() {
+option::template defaultTmplOpts() {
 	stringField "goVersion" "1.14.2"
 	stringField "platform" localOs
 	stringField "arch" localArch
 	stringField "goBuildFlags" ""
 	stringField "binaryName" ""
 	stringField "buildDir" "build"
 	stringField "golangciLintVersion" "latest"
-	stringField "cgoEnabled" "1"
 }
 
-string goBuildCommand(string package, variadic option::template args) {
+string goBuildCommand(string package, variadic option::template tmplOpts) {
 	template <<~EOM
-		CGO_ENABLED={{.cgoEnabled}} go build {{.goBuildFlags}} -o /out/{{.binaryName}} {{.package}}
+		go build {{.goBuildFlags}} -o /out/{{.binaryName}} {{.package}}
 	EOM with option {
-		defaultArgs
-		args
+		defaultTmplOpts
+		tmplOpts
 		stringField "package" package
 	}
 }
 
-fs _runBuild(fs input, string package, variadic option::template args) {
+fs base(variadic option::template tmplOpts) {
 	image string {
-		builderImage args
+		builderImage tmplOpts
 	} with resolve
+	# need to install ssh for git/private go mod deps
+	run "apt-get update && apt-get -y install ssh"
+}
+
+# Builds a go binary.
+#
+# @param input a filesystem containing go source files.
+# @param package a relative path for the package to build.
+# @return a filesystem containing only the built binary.
+fs build(fs input, string package) {
+	buildWithOptions input package nullTemplateOpts nullRunOpts
+}
+
+# nullTemplateOpts is just an explicitly named empty option list
+option::template nullTemplateOpts() {
```

Comment:
Is this a workaround?

Reply:
No, not actually necessary, I thought it just improved readability.  But looking again I think we can just use `option::template{}`, I was put off that early because syntax highlighting was not working with it, but probably just a bug in the hlb.iro spec.  It seems to work fine.